pkg/controller/atlasdeployment/atlasdeployment_controller.go

```diff
@@ -162,6 +162,10 @@ func (r *AtlasDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	wasDeleted := !atlasDeployment.GetDeletionTimestamp().IsZero()
 	existsInAtlas := deploymentInAtlas != nil
 
+	if existsInAtlas && fmt.Sprintf("%T", deploymentInAKO) != fmt.Sprintf("%T", deploymentInAtlas) {
```

Comment:
No, please don't switch on string representations of types, this is not right. A low level `%T` language type primitive must not be used to assert high level business logic. You can accomplish this using different techniques.

I.e. you can extend the `translation.Deployment` interface with a `IsServerless()` method. In the case of `NewDeployment` it returns it based on the spec and in the case of the `GetDeployment` service it returns it based whether an underlying serverless or regular deployment was found.

Reply:
the assertion would then change to something like:
```
	if existsInAtlas && deploymentInAKO.IsServerless() != deploymentInAtlas.IsServerless() {
```